src/main/java/com/neonlab/product/apis/DeleteProductApi.java

```diff
@@ -20,12 +20,17 @@ public class DeleteProductApi {
     @Autowired
     private ProductService productService;
 
-    public ApiOutput<?>deleteProductApi(ProductDeleteReq productDeleteReq){
+    public ApiOutput<?>deleteProductApi(ProductDeleteReq productDeleteReq, boolean removeWholeProduct){
```

Comment:
Please move the `removeWholeProduct` in the `ProductDeleteReq`. Also rename it to `deleteProduct`

cc- @sjha24